### PR TITLE
Refactor ledger-test-tool

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: ledger-test-tool/tool/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+            "dev-mode=no",
+        ]),
+    ],
+    test_tool_args = conformance_test_tool_args_base + ["--include=TLSAtLeastOnePointTwoIT"],
+) if not is_windows else None
+
+# conformance tests for preview and dev
+#  those test the conformance test it-self for preview/dev features and the patched canton jar
+conformance_test(
+    name = "conformance-test",
```
Comment (mziolekda) [inline ledger-test-tool/tool/BUILD.bazel:242]: same name used twice

File: ledger-test-tool/tool/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+# this tests TLSAtLeastOnePointTwoIT
+conformance_test(
+    name = "conformance-test-tls1.2-or-newer",
+    extra_data = conformance_test_extra_data_base + ["//canton:canton_deploy.jar"],
+    extra_runner_args = ["7000"],
+    lf_versions = ["latest"],
+    ports = conformance_test_ports,
+    runner = "@//bazel_tools/client_server/runner_with_health_check",
+    server = ":canton-test-runner-with-dependencies",
+    server_args = conformance_server_args_base + [
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply ledger-test-tool/tool/BUILD.bazel:230]: I tried to stay the closer possible to the original test.
It seems that 1.2 is the default, as the original test did not set the the server-protocol and it is actually green without for me like that.

```
//ledger-test-tool/tool:conformance-test-tls1.2-or-newer-1.15            PASSED in 72.8s
```
 
Note that we have `minimum-server-protocol-version=TLSv1.3` in the test just above.

File: ledger-test-tool/tool/BUILD.bazel (hunk 1 above)
Comment (remyhaemmerle-da) [reply ledger-test-tool/tool/BUILD.bazel:242]: conformance_test suffix the test with the LF version, so this actually does not conflict. 

```
//ledger-test-tool/tool:conformance-test-1.15                            PASSED in 462.9s
//ledger-test-tool/tool:conformance-test-1.dev                           PASSED in 444.0s
//ledger-test-tool/tool:conformance-test-tls1.2-or-newer-1.15            PASSED in 72.8s
//ledger-test-tool/tool:test-extract                                     PASSED in 1.8s
```

Anyway, do you prefer I use something like `conformance-test-dev` ?

File: ledger-test-tool/tool/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        "-C",
+        ",".join([
+            "dev-mode=yes",
+            "canton.domains.test_domain.init.domain-parameters.protocol-version=dev",
+        ]),
+    ],
+    tags = ["dev-canton-test"],
+    test_tool_args = conformance_test_tool_args_base + [
+        "--exclude=" + ",".join([
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply ledger-test-tool/tool/BUILD.bazel]: b73b7959d604659528915a23a247bc6f2c5ce7f4

File: ledger-test-tool/tool/BUILD.bazel (hunk 2 above)
[the remark this answers is not in the document]
Comment (mziolekda) [reply ledger-test-tool/tool/BUILD.bazel:232]: yes it seems indeed to be the default :facepalm:

**Pull request merged**